include/grpc++/impl/codegen/call.h

```diff
@@ -94,11 +95,65 @@ inline grpc_metadata* FillMetadataArray(
   return metadata_array;
 }
 
+/// Per-message options for all operation types
+class MessageOptions {
+ public:
+  MessageOptions()
+      : msg_deadline_(
+            g_core_codegen_interface->gpr_inf_future(GPR_CLOCK_REALTIME)) {}
+  gpr_timespec deadline() const { return msg_deadline_; }
+  template <class T>
+  void set_deadline(const T& deadline) {
```

Comment:
It doesn't get propagated through a chain of RPCs, but it does have to get passed through multiple layers of the stack and will trigger asynchronously (by materializing an alarm at the appropriate time that pushes a "quell" op). Unless there's a strong reason otherwise, I feel that consistency weighs in favor of a deadline.